packages/host/app/components/operator-mode/profile-info-popover.gts

```diff
@@ -31,59 +36,187 @@ export default class ProfileInfoPopover extends Component<ProfileInfoPopoverSign
-        data-test-signout-button
-      >
-        Sign out
-      </BoxelButton>
+      <div class='credit-info' data-test-credit-info>
+        <div class='info-group'>
+          <span class='label'>Membership Tier</span>
+          <span class='value' data-test-membership-tier>{{this.plan}}</span>
+        </div>
+        {{#if this.isChangePlanDisplayed}}
+          <BoxelButton
+            @kind='secondary-light'
+            @size='small'
+            data-test-change-plan-button
+            {{on 'click' @toggleProfileSettings}}
+          >Change plan</BoxelButton>
+        {{/if}}
+        <div class='info-group'>
+          <span class='label'>Monthly Credit</span>
+          <span
+            class={{cn 'value' out-of-credit=this.isOutOfCredit}}
+            data-test-monthly-credit
+          >{{this.monthlyCredit}}</span>
+        </div>
+        {{#if this.isOutOfCredit}}
+          <span class='out-of-credit-info' data-test-out-of-credit>You have used
+            all of credits from the free trial, please upgrade your plan.</span>
+        {{/if}}
+        <div class='info-group'>
+          <span class='label'>Additional Credit</span>
+          <span class='value' data-test-additional-balance>{{if
+              this.isFreePlan
+              'Upgrade to Enable'
+              this.creditsAvailableInBalance
+            }}</span>
+        </div>
+      </div>
+      {{#if this.isFreePlan}}
+        <BoxelButton
+          {{on 'click' @toggleProfileSettings}}
+          style={{cssVar
+            boxel-button-text-color='var(--boxel-dark)'
+            boxel-button-color='var(--boxel-teal)'
+            boxel-button-border='1px solid var(--boxel-teal)'
+          }}
+          @kind='primary'
+          data-test-choose-plan-button
+        >
+          Choose Plan
+        </BoxelButton>
+      {{/if}}
     </div>
   </template>
+
+  @service private declare realmServer: RealmServerService;
+
+  private fetchCreditInfo = trackedFunction(this, async () => {
+    return await this.realmServer.fetchCreditInfo();
+  });
+
+  private get plan() {
+    return this.fetchCreditInfo.value?.plan;
+  }
+
+  private get creditsIncludedInPlanAllowance() {
+    return this.fetchCreditInfo.value?.creditsIncludedInPlanAllowance;
+  }
+
+  private get creditsUsedInPlanAllowance() {
+    return this.fetchCreditInfo.value?.creditsUsedInPlanAllowance;
+  }
+
+  private get creditsAvailableInBalance() {
+    return this.fetchCreditInfo.value?.creditsAvailableInBalance;
+  }
+
+  private get monthlyCredit() {
+    return this.creditsUsedInPlanAllowance != undefined &&
+      this.creditsIncludedInPlanAllowance != undefined
+      ? `${
+          this.creditsIncludedInPlanAllowance - this.creditsUsedInPlanAllowance
+        } of ${this.creditsIncludedInPlanAllowance} left`
+      : '';
+  }
+
+  private get isOutOfCredit() {
+    return (
+      this.isFreePlan &&
+      this.creditsUsedInPlanAllowance === this.creditsIncludedInPlanAllowance
```

Comment:
It's more robust to say `this.creditsUsedInPlanAllowance >= this.creditsIncludedInPlanAllowance`. There is a slight chance there can be a little overspend (in case the background job that will write the consumption will be late, and the logic that calculates whether user has enough credit to perform an AI action will have outdated data)